subsys/metrics/src/main/java/org/commonjava/indy/metrics/sigar/FilesystemMetricsSet.java

```diff
@@ -0,0 +1,138 @@
+
+package org.commonjava.indy.metrics.sigar;
+
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.Metric;
+import com.codahale.metrics.MetricSet;
+import org.hyperic.sigar.FileSystem;
+import org.hyperic.sigar.FileSystemUsage;
+import org.hyperic.sigar.Sigar;
+import org.hyperic.sigar.SigarException;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class FilesystemMetricsSet implements MetricSet
+{
+    @Override
+    public Map<String, Metric> getMetrics() {
+        final Map<String, Metric> gauges = new HashMap<String, Metric>();
+        Sigar sigar = SigarService.getSigar();
+        try {
+            FileSystem[] fileSystems = sigar.getFileSystemList();
+            for (FileSystem fs : fileSystems) {
+                if (fs.getType() != FileSystem.TYPE_LOCAL_DISK)
+                    continue;
+                createGaugesForFilesystem(gauges, fs);
+            }
+        } catch (SigarException e) {
+            //
+        }
+        return gauges;
+    }
+
+    private void createGaugesForFilesystem( Map<String, Metric> gauges, FileSystem fs) throws SigarException {
+        final String fsDirName = fs.getDirName();
+        String fsDevName = fs.getDevName();
+        String fsName = fsDevName.substring(fsDevName.lastIndexOf('/') + 1);
+
+         gauges.put("fileSystem." + fsName + ".diskQueue", new Gauge<Double>() {
```

Comment:
For things like this that are related to the underlying system, we need to use a common base name, like `system.` in front of all names. That will give us more control over turning these on/off later when we have fine-grained configuration controls.